pytorch_lightning/accelerators/accelerator.py

```diff
@@ -79,15 +79,14 @@ def setup_environment(self) -> None:
         """
         self.training_type_plugin.setup_environment()
 
-    def setup(self, trainer: 'pl.Trainer', model: 'pl.LightningModule') -> None:
+    def setup(self, trainer: 'pl.Trainer') -> None:
         """
         Setup plugins for the trainer fit and creates optimizers.
 
         Args:
             trainer: the trainer instance
-            model: the LightningModule
         """
-        self.setup_training_type_plugin(model)
+        self.setup_training_type_plugin()
```

Comment:
This would be a BC for people who built Plugins.

Reply:
Unless we are super certain no one has built plugins, I'd like to see some form of deprecation path here!

Reply:
I know, I included it in the breaking changes at the top.

The model has been connected at this point, so I don't see any good argument to keep the parameter.
And none of our plugins were making special use of the argument.

Reply:
> Unless we are super certain no one has built plugins, I'd like to see some form of deprecation path here!

Previous PRs have already implemented breaking changes for the TTP API (https://github.com/PyTorchLightning/pytorch-lightning/pull/8328 for example). As far as I know, we still consider it experimental/beta.

Reply:
I think @kandluis et al. have recently hinted at it that they have plugin implementations. 
Let's try to remove the beta state in 1.5 and make it stable? :) 
We might need #7324 first though :)

Reply:
This argument update is not strictly necessary for me to finish #8498

I chose to remove it here because there doesn't seem to be a reason to keep the argument, as no plugin uses it and the usage pattern should always follow:

```
plugin.connect(model)  # sets the `self.model` reference in in the plugin
plugin.setup()
```

If anybody has a good reason for it, we can keep it

Reply:
```suggestion
        self.setup_training_type_plugin(_)
```
for compatibility?